### PR TITLE
qb_chain: 2.2.2-1 in 'noetic/distribution.yaml' [bloom]

### PR DESCRIPTION
**Pull request by umbertofontana93**
Increasing version of package(s) in repository `qb_chain` to `2.2.2-1`:

- upstream repository: https://bitbucket.org/qbrobotics/qbchain-ros.git
- release repository: https://bitbucket.org/qbrobotics/qbchain-ros-release.git
- distro file: `noetic/distribution.yaml`
- bloom version: `0.10.7`
- previous version for package: `null`

## qb_chain

- No changes

## qb_chain_control

- No changes

## qb_chain_controllers

- No changes

## qb_chain_description

- No changes

## qb_chain_msgs

```
* Fix version tag in some packages.
```

### CODE REVIEW

**Review by audrow — APPROVED** [review]

**Comment by ivanpauno** [comment]
@umbertofontana93 could you resolve conflicts?

**Comment by sloretz** [comment]
[Holding for Noetic sync](https://discourse.ros.org/t/preparing-for-noetic-sync-2021-09-06/)

**Comment by tfoote** [comment]
Unfortunately this now conflicts on merge with #30601 please rebase and comment to let us know when it's ready.

**Review by tfoote — CHANGES_REQUESTED** [review]
This is not adding qb_chain like the description, it's replacing qb_move with qb_chain

File: noetic/distribution.yaml [hunk 1]
```diff
@@ -5969,8 +5969,76 @@ repositories:
       test_pull_requests: true
       type: git
       url: https://github.com/ros-visualization/python_qt_binding.git
-      version: melodic-devel
+      version: noetic-devel
```
Comment (tfoote) [inline noetic/distribution.yaml:5972]: This is an unrelated change that is pointing to a non-extant branch. Please don't make unrelated changes.

File: noetic/distribution.yaml [hunk 2]
```diff
@@ -5959,7 +5959,7 @@ repositories:
     doc:
       type: git
       url: https://github.com/ros-visualization/python_qt_binding.git
-      version: melodic-devel
+      version: noetic-devel
```
Comment (tfoote) [inline noetic/distribution.yaml:5962]: This is an unrelated change that is pointing to a non-extant branch. Please don't make unrelated changes.

File: noetic/distribution.yaml [hunk 3]
```diff
@@ -5969,8 +5969,76 @@ repositories:
...
+      tags:
+        release: release/noetic/{package}/{version}
+      url: https://bitbucket.org/qbrobotics/qbchain-ros-release.git
+      version: 2.2.2-1
+    source:
+      type: git
+      url: https://bitbucket.org/qbrobotics/qbchain-ros.git
+      version: production-noetic
+    status: developed
+  qb_device:
```
Comment (tfoote) [inline noetic/distribution.yaml:5995]: This pull request relates to qb_chain. 

I've suggested that you stack your commits so that they can be merged sequentially. However I cannot see the sequence in which to merge these. They all seem to cross reference. 

Please comment and clarify how you expect them to be merged. Ideally each one will have one commit inserting one packages. If it will be easier for you. You can do each one at a time and use bloom-release with `-p` option to regenerate the additional requests after the merged PR is completed, if the git history is getting too complicated.

**Review by tfoote — CHANGES_REQUESTED** [review]
I've opened a new PR #30766  with what I think you want. Can you confirm that's what you want and we can close this in favor of that?

Below is what I did which you can redo after that one is merged to get a clean patch for your next package.

<details>


```
$ bloom-release -p qb_chain --rosdistro noetic
ROS Distro index file associate with commit '50ae5e17bb69381fa9a991ef4afbd3872874992b'
New ROS Distro index url: 'https://raw.githubusercontent.com/ros/rosdistro/50ae5e17bb69381fa9a991ef4afbd3872874992b/index-v4.yaml'
Specified repository 'qb_chain' is not in the distribution file located at 'https://raw.githubusercontent.com/ros/rosdistro/50ae5e17bb69381fa9a991ef4afbd3872874992b/noetic/distribution.yaml'
Did you mean one of these: 'cob_hand'?
Could not determine release repository url for repository 'qb_chain' of distro 'noetic'
You can continue the release process by manually specifying the location of the RELEASE repository.
To be clear this is the url of the RELEASE repository not the upstream repository.
For release repositories on GitHub, you should provide the `https://` url which should end in `.git`.
Here is the url for a typical release repository on GitHub: https://github.com/ros-gbp/rviz-release.git
==> Looking for a release of this repository in a different distribution...
A different distribution, 'melodic', released this repository.
Release repository url [https://bitbucket.org/qbrobotics/qbchain-ros-release.git]: 
==> Fetching 'qb_chain' repository from 'https://bitbucket.org/qbrobotics/qbchain-ros-release.git'
Cloning into '/tmp/tmpams88omc'...
remote: Enumerating objects: 3825, done.
remote: Counting objects: 100% (1382/1382), done.
remote: Compressing objects: 100% (959/959), done.
remote: Total 3825 (delta 370), reused 1299 (delta 300), pack-reused 2443
Receiving objects: 100% (3825/3825), 2.42 MiB | 8.90 MiB/s, done.
Resolving deltas: 100% (1158/1158), done.
==> Generating pull request to distro file located at 'https://raw.githubusercontent.com/ros/rosdistro/50ae5e17bb69381fa9a991ef4afbd3872874992b/noetic/distribution.yaml'
Would you like to add documentation information for this repository? [Y/n]? 
==> Looking for a doc entry for this repository in a different distribution...
Using defaults from the doc entry of distribution 'melodic'.
Please enter your repository information for the doc generation job.
This information should point to the repository from which documentation should be generated.
VCS Type must be one of git, svn, hg, or bzr.
VCS type [git]: 
VCS url [https://bitbucket.org/qbrobotics/qbchain-ros.git]: 
VCS version must be a branch, tag, or commit, e.g. master or 0.1.0
VCS version [production-melodic]: 
Would you like to add source information for this repository? [Y/n]? ^C
Received 'KeyboardInterrupt', aborting.
tfoote@snowman5:~ Last: [0] (43s Seconds)
$ bloom-release -p qb_chain --rosdistro noetic
ROS Distro index file associate with commit '50ae5e17bb69381fa9a991ef4afbd3872874992b'
New ROS Distro index url: 'https://raw.githubusercontent.com/ros/rosdistro/50ae5e17bb69381fa9a991ef4afbd3872874992b/index-v4.yaml'
Specified repository 'qb_chain' is not in the distribution file located at 'https://raw.githubusercontent.com/ros/rosdistro/50ae5e17bb69381fa9a991ef4afbd3872874992b/noetic/distribution.yaml'
Did you mean one of these: 'cob_hand'?
Could not determine release repository url for repository 'qb_chain' of distro 'noetic'
You can continue the release process by manually specifying the location of the RELEASE repository.
To be clear this is the url of the RELEASE repository not the upstream repository.
For release repositories on GitHub, you should provide the `https://` url which should end in `.git`.
Here is the url for a typical release repository on GitHub: https://github.com/ros-gbp/rviz-release.git
==> Looking for a release of this repository in a different distribution...
A different distribution, 'melodic', released this repository.
Release repository url [https://bitbucket.org/qbrobotics/qbchain-ros-release.git]: 
==> Fetching 'qb_chain' repository from 'https://bitbucket.org/qbrobotics/qbchain-ros-release.git'
Cloning into '/tmp/tmp2nk7kitb'...
Receiving objects: 100% (3825/3825), 2.42 MiB | 8.93 MiB/s, done.
Resolving deltas: 100% (1158/1158), done.
==> Generating pull request to distro file located at 'https://raw.githubusercontent.com/ros/rosdistro/50ae5e17bb69381fa9a991ef4afbd3872874992b/noetic/distribution.yaml'
Would you like to add documentation information for this repository? [Y/n]? 
==> Looking for a doc entry for this repository in a different distribution...
Using defaults from the doc entry of distribution 'melodic'.
Please enter your repository information for the doc generation job.
This information should point to the repository from which documentation should be generated.
VCS Type must be one of git, svn, hg, or bzr.
VCS type [git]: 
VCS url [https://bitbucket.org/qbrobotics/qbchain-ros.git]: 
VCS version must be a branch, tag, or commit, e.g. master or 0.1.0
VCS version [production-melodic]: production-noetic
Would you like to add source information for this repository? [Y/n]? 
==> Looking for a source entry for this repository in a different distribution...
Using defaults from the source entry of distribution 'melodic'.
Please enter information which points to the active development branch for this repository.
This information is used to run continuous integration jobs and for developers to checkout from.
VCS Type must be one of git, svn, hg, or bzr.
VCS type [git]: 
VCS url [https://bitbucket.org/qbrobotics/qbchain-ros.git]: 
VCS version must be a branch, tag, or commit, e.g. master or 0.1.0
VCS version [production-melodic]: production-noetic
Since you are on github we can add a job to run your tests on each pull request.If you would like to turn this on please see http://wiki.ros.org/buildfarm/Pull%20request%20testing for more information. There is more setup required to setup the hooks correctly. 
Would you like to turn on pull request testing? [y/N]? 
Would you like to add a maintenance status for this repository? [Y/n]? 
Please enter a maintenance status.
Valid maintenance statuses:
- developed: active development is in progress
- maintained: no new development, but bug fixes and pull requests are addressed
- unmaintained: looking for new maintainer, bug fixes and pull requests will not be addressed
- end-of-life: should not be used, will disappear at some point
Status: developed
You can also enter a status description.
This is usually reserved for giving a reason when a status is 'end-of-life'.
Status Description [press Enter for no change]: 
Unified diff for the ROS distro file located at '/tmp/tmp7ryic53j/qb_chain-2.2.2-1.patch':
--- 50ae5e17bb69381fa9a991ef4afbd3872874992b/noetic/distribution.yaml
+++ 50ae5e17bb69381fa9a991ef4afbd3872874992b/noetic/distribution.yaml
@@ -5976,6 +5976,27 @@
       url: https://github.com/ros-visualization/python_qt_binding.git
       version: melodic-devel
     status: maintained
+  qb_chain:
+    doc:
+      type: git
+      url: https://bitbucket.org/qbrobotics/qbchain-ros.git
+      version: production-noetic
+    release:
+      packages:
+      - qb_chain
+      - qb_chain_control
+      - qb_chain_controllers
+      - qb_chain_description
+      - qb_chain_msgs
+      tags:
+        release: release/noetic/{package}/{version}
+      url: https://bitbucket.org/qbrobotics/qbchain-ros-release.git
+      version: 2.2.2-1
+    source:
+      type: git
+      url: https://bitbucket.org/qbrobotics/qbchain-ros.git
+      version: production-noetic
+    status: developed
   qb_move:
     doc:
       type: git
==> Checking on GitHub for a fork to make the pull request from...
==> Using this fork to make a pull request from: tfoote/rosdistro
==> Cloning tfoote/rosdistro...
==> mkdir -p rosdistro
==> git init
Initialized empty Git repository in /tmp/xcj1_y7i/rosdistro/.git/
Pull Request Title: qb_chain: 2.2.2-1 in 'noetic/distribution.yaml' [bloom]
Pull Request Body : 
Increasing version of package(s) in repository `qb_chain` to `2.2.2-1`:

- upstream repository: https://bitbucket.org/qbrobotics/qbchain-ros.git
- release repository: https://bitbucket.org/qbrobotics/qbchain-ros-release.git
- distro file: `noetic/distribution.yaml`
- bloom version: `0.10.7`
- previous version for package: `null`

## qb_chain

- No changes

## qb_chain_control

- No changes

## qb_chain_controllers

- No changes

## qb_chain_description

- No changes

## qb_chain_msgs

```
* Fix version tag in some packages.
```

Open a pull request from 'tfoote/rosdistro:bloom-qb_chain-0' into 'ros/rosdistro:master'?
Continue [Y/n]? 
==> git checkout -b bloom-qb_chain-0
Switched to a new branch 'bloom-qb_chain-0'
==> Pulling latest rosdistro branch
remote: Enumerating objects: 163474, done.
remote: Counting objects: 100% (36/36), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 163474 (delta 16), reused 31 (delta 14), pack-reused 163438
Receiving objects: 100% (163474/163474), 99.64 MiB | 10.43 MiB/s, done.
Resolving deltas: 100% (98531/98531), done.
From https://github.com/ros/rosdistro
 * branch            master     -> FETCH_HEAD
==> git reset --hard 50ae5e17bb69381fa9a991ef4afbd3872874992b
HEAD is now at 50ae5e17b qb_device: 2.2.2-1 in 'noetic/distribution.yaml' [bloom] (#30664)
==> Writing new distribution file: noetic/distribution.yaml
==> git add noetic/distribution.yaml
==> git commit -m "qb_chain: 2.2.2-1 in 'noetic/distribution.yaml' [bloom]"
[bloom-qb_chain-0 bfc99ac4b] qb_chain: 2.2.2-1 in 'noetic/distribution.yaml' [bloom]
 1 file changed, 21 insertions(+)
==> Pushing changes to fork
Enumerating objects: 7, done.
Counting objects: 100% (7/7), done.
Delta compression using up to 8 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (4/4), 513 bytes | 513.00 KiB/s, done.
Total 4 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
remote: 
remote: Create a pull request for 'bloom-qb_chain-0' on GitHub by visiting:
remote:      https://github.com/tfoote/rosdistro/pull/new/bloom-qb_chain-0
remote: 
To https://github.com/tfoote/rosdistro.git
 * [new branch]          bloom-qb_chain-0 -> bloom-qb_chain-0
<== Pull request opened at: https://github.com/ros/rosdistro/pull/30766
```
</details>

File: noetic/distribution.yaml (hunk 1 above)
Comment (umbertofontana93) [reply noetic/distribution.yaml:5972]: yes it is what I want

File: noetic/distribution.yaml (hunk 2 above)
Comment (umbertofontana93) [reply noetic/distribution.yaml:5962]: yes it is what I want

**Comment by tfoote** [comment]
superceded
